lib/opus/pipeline/stage/skip.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Opus.Pipeline.Stage.Skip do
+  @moduledoc ~S"""
+  The skip stage is meant to halt the pipeline with no error if the given condition is true.
+  This stage must be called with an `if` option, in order to decide if the pipeline is going to be
+  halted or not.
+
+  When the given conditional is `true`, the pipeline will return {:ok, :skipped} and all the following
+  steps will be skipped.
+
+  ```
+  defmodule CreateUserPipeline do
+    use Opus.Pipeline
+
+    skip if: :user_exists?
+    step :persist_user
+  end
+  ```
+
+  In this example, if the `user_exists?` implementation returns `true`, then the next step `persist_user`
+  is not going to be called. If `false`, then Opus will keep following to the next stages.
+  """
+
+  alias Opus.Pipeline.Stage
+
+  @behaviour Stage
+
+  def run({module, type, [if: func], opts} = stage, input) do
+    case Stage.maybe_run({module, type, nil, opts |> put_in([:if], func)}, input) do
+      ret when ret == :stage_skipped ->
```

Comment:
```elixir  
def run({module, type, [if: func], opts}, input) do
    case Stage.maybe_run({module, type, nil, opts |> put_in([:if], func)}, input) do
      :stage_skipped -> {:halt, :skipped}
      _ -> {:cont, input}
    end
  end
```

1. The guard is not needed here
2. When the conditional returns false (and we don't skip) you can return `{:cont, input}` so that the pipeline continues.

Reply:
Totally, thanks for noticing! Just changed!